zip-0316.rst

```diff
@@ -3,11 +3,556 @@
+  range of potential future features. Some examples might include Layer 2
+  features, cross-chain interoperability and bridging, and decentralized
+  exchange.
+- The standard should work well for Zcash today and upcoming potential
+  upgrades, and also anticipate even broader use cases down the road such
+  as cross-chain functionality.
+
+
+Requirements
+============
+
+Overview
+--------
+
+Unified Addresses specify multiple methods for payment to a Recipient's Wallet.
+The Sender's Wallet can then non-interactively select the method of payment.
+
+Importantly, any wallet can support Unified Addresses, even when that wallet
+only supports a subset of payment methods for receiving and/or sending.
+
+Despite having some similar characteristics, the Unified Address standard is
+orthogonal to Payment Request URIs [#zip-0321]_ and similar schemes, and the
+Unified Address format is likely to be incorporated into such schemes as a new
+address type.
+
+Concepts
+--------
+
+Wallets follow a model *Interaction Flow* as follows:
+
+1. A Recipient *generates* an Address.
+2. The Recipient wallet or human user encodes the address, and
+   *distributes* this Address Encoding through mechanisms which may be
+   “out-of-band” (example: they spray paint a QR Code on a sign) or
+   they may be more or less “in-band” (example: they include a string
+   encoding of a ``Reply-To`` address in an encrypted memo following a
+   common standard).
+3. A Sender wallet or user *imports* the Address Encoding through any of
+   a variety of mechanisms (QR Code scanning, Payment URIs, cut-and-paste,
+   or “in-band” protocols like ``Reply-To`` memos). This includes decoding
+   and validity checks.
+4. (Perhaps later in time) the Sender wallet executes a transfer of ZEC
+   (or other assets or future protocol state changes) to the Address.
+
+These steps are a funnel: encodings of the same Address may be distributed
+zero or more times through different means. Zero or more Senders may import
+addresses. Zero or more of those may execute a Transfer. A single Sender may
+execute multiple Transfers over time from a single import.
+
+[TODO: examples]
+
+Addresses
+---------
+
+A Unified Address (or UA for short) combines one or more Receivers.
+
+When new Transport Protocols are introduced to the Zcash protocol after
+Unified Addresses are standardized, those should introduce new Receiver Types
+but *not* different address types outside of the UA standard. There needs
+to be a compelling reason to deviate from the standard, since the benefits
+of UA come precisely from their applicability across all new protocol
+upgrades.
+
+Receivers
+---------
+
+Every Wallet must anticipate and properly parse a UA with any unknown
+arbitrary Receiver Type.
+
+When Transferring to a valid UA, a Sender must behave as if any unknown
+Receiver Type is simply not present for the purposes of the transfer.
+
+A Wallet may process unknown Receiver Types by indicating to the user
```

Comment:
What should a parser or a parsing wallet do in the presence of a sound UA that contains an invalid receiver component?
Should it treat the invalid receiver the same way as unknown ones? should that render the rest of the receivers components invalid as well?

Reply:
The decoding algorithm now says that the whole UA MUST be rejected in this case.